ast/src/parsed/mod.rs

```diff
@@ -633,6 +635,71 @@ pub enum BinaryOperator {
     Greater,
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub enum BinaryOperatorAssociativity {
+    Left,
+    Right,
+    RequireParentheses,
+}
+
+const UNARY_EXPRESSION_PRIORITY_OFFSET: ExpressionPrecedence = 0; // All unary operators have the same precedence.
+const BINARY_EXPRESSION_PRIORITY_OFFSET: ExpressionPrecedence = 10; // Binary operators will have a lower precedence than unary operators.
+
+impl UnaryOperator {
+    pub fn precedence(&self) -> ExpressionPrecedence {
+        use UnaryOperator::*;
+        UNARY_EXPRESSION_PRIORITY_OFFSET
+            + match self {
+                Minus | LogicalNot => 1,
```

Comment:
I think these should be swapped: https://github.com/powdr-labs/powdr/blob/main/parser/src/powdr.lalrpop#L555